src/libraries/Fuzzing/DotnetFuzzing/Assert.cs

```diff
@@ -45,4 +45,24 @@ static void Throw(ReadOnlySpan<T> expected, ReadOnlySpan<T> actual)
             throw new Exception($"Expected={expected[diffIndex]} Actual={actual[diffIndex]} at index {diffIndex}");
         }
     }
+
+    public static void Throws<T, TState>(Action<TState> action, TState state)
+        where T : Exception
+        where TState : allows ref struct
+    {
+        try
+        {
+            action(state);
+        }
+        catch (T)
+        {
+            return;
```

Comment:
Consider returning the caught exception, since that's what existing `Assert.Throws*` methods typically do.